### ISSUE TITLE
`Multiple resources`-exception

### ISSUE THREAD

**Issue by osirisinferi** [iBotPeaches/Apktool#1087]
L.S.,

When running the command `apktool d Mail+nonprime.apk`, I get the following output/error:

```
I: Using Apktool 2.0.2 on Mail+nonprime.apk
I: Loading resource table...
Exception in thread "main" brut.androlib.AndrolibException: Multiple resources: spec=0x7f080000 dimen/recipient_primary_text_size, config=-xhdpi-v4
	at brut.androlib.res.data.ResConfig.addResource(ResConfig.java:63)
	at brut.androlib.res.data.ResConfig.addResource(ResConfig.java:56)
	at brut.androlib.res.decoder.ARSCDecoder.readEntry(ARSCDecoder.java:223)
	at brut.androlib.res.decoder.ARSCDecoder.readConfig(ARSCDecoder.java:191)
	at brut.androlib.res.decoder.ARSCDecoder.readType(ARSCDecoder.java:159)
	at brut.androlib.res.decoder.ARSCDecoder.readPackage(ARSCDecoder.java:116)
	at brut.androlib.res.decoder.ARSCDecoder.readTable(ARSCDecoder.java:78)
	at brut.androlib.res.decoder.ARSCDecoder.decode(ARSCDecoder.java:47)
	at brut.androlib.res.AndrolibResources.getResPackagesFromApk(AndrolibResources.java:544)
	at brut.androlib.res.AndrolibResources.loadMainPkg(AndrolibResources.java:63)
	at brut.androlib.res.AndrolibResources.getResTable(AndrolibResources.java:55)
	at brut.androlib.Androlib.getResTable(Androlib.java:65)
	at brut.androlib.ApkDecoder.setTargetSdkVersion(ApkDecoder.java:197)
	at brut.androlib.ApkDecoder.decode(ApkDecoder.java:96)
	at brut.apktool.Main.cmdDecode(Main.java:165)
	at brut.apktool.Main.main(Main.java:81)
```

According to some other bugs, this has something to do inside the APK and could be fixed in ApkTool, but this isn't something I'm thát familiar with.

APK can be found at: http://sourceforge.net/p/kk42one442s55/trunk/131/tree/trunk/system/app/Mail%2Bnonprime.apk

**Comment by iBotPeaches** [comment]
So this has to be related to #644 #1072 or #1045 . This is another HTC problem which literally makes zero sense to me. Duplicate resources means a collision occurs in the Configuration flags during resource decoding.

Usually using `aapt d configurations` or `aapt d resources` I can find the improper configuration and fix it from there, but you can see #644 is a 2 year old bug report, and this issue follows suit with the same problem.

Here I am dumping resources and grepping for the resourceId of `0x7f100000`, which you can see is the cause of the crash. Here is the output.

![bug](https://cloud.githubusercontent.com/assets/611784/11323707/bbbc652a-90df-11e5-99bf-48637e40a6a8.png)

This has to be a misunderstanding on my part, because there is multiple configurations of `xhdpi-v4` with the same resource. You can see towards the end it transitions to `xxhdpi-v4`, but the issue is earlier where multiple resources are encountered.

I'm not sure how HTC continues to do this (Probably with some modified `aapt` or how their OEM builds of Android handles this), but this is a long running bug with no solution in sight :/ 

Any research or help would be greatly appreciated.

**Comment by osirisinferi** [comment]
But is this really something that should be handled with a fatal error?

Can't it be handled with a warning or perhaps a query to the user or something? Perhaps something like: "WARNING: Multiple resources found: how do you want to handle this? a) Quit; b) Discard everything but the first resource; c) Let me choose one of the multiple resources found".

As far as I'm concerend, all but the first of those `xhdpi-v4`-resources are going to /dev/null. Get rid of them.. But I've got no idea what the consequences are for the rest of the decoding :P

By the way, the same goes for xxhdpi-v4: there are also 5 configs listed..
